internal/pkg/githubapi/github.go

```diff
@@ -90,10 +98,43 @@ func HandlePREvent(eventPayload *github.PullRequestEvent, ghPrClientDetails GhPr
 		SetCommitStatus(ghPrClientDetails, "pending")
 		wasCommitStatusSet = true
 		botIdentity, _ := GetBotGhIdentity(mainGithubClientPair.v4Client, ctx)
-		err := MimizeStalePrComments(ghPrClientDetails, mainGithubClientPair.v4Client, botIdentity)
+		err = MimizeStalePrComments(ghPrClientDetails, mainGithubClientPair.v4Client, botIdentity)
 		if err != nil {
 			prHandleError = err
-			log.Errorf("Failed to minimize stale comments: err=%s\n", err)
+			ghPrClientDetails.PrLogger.Errorf("Failed to minimize stale comments: err=%s\n", err)
+		}
+		if config.CommentArgocdDiffonPR {
+			componentPathList := generateListOfChangedComponentPaths(ghPrClientDetails, config)
+			// TODO handle error
+			noDiffsAndErrorsPR, diffOfChangedComponents, err := argocd.GenerateDiffOfChangedComponents(ctx, componentPathList, ghPrClientDetails.Ref, ghPrClientDetails.RepoURL)
+			if err != nil {
+				prHandleError = err
+				ghPrClientDetails.PrLogger.Errorf("Failed to get ArgoCD diff information: err=%s\n", err)
+			} else {
+				ghPrClientDetails.PrLogger.Debugf("Successfully got ArgoCD diff\n")
+				if noDiffsAndErrorsPR {
+					ghPrClientDetails.PrLogger.Debugf("ArgoCD diff is empty, this PR will not change cluster state\n")
+					prLables, resp, err := ghPrClientDetails.GhClientPair.v3Client.Issues.AddLabelsToIssue(ghPrClientDetails.Ctx, ghPrClientDetails.Owner, ghPrClientDetails.Repo, *eventPayload.PullRequest.Number, []string{"noop"})
+					prom.InstrumentGhCall(resp)
+					if err != nil {
+						ghPrClientDetails.PrLogger.Errorf("Could not label GitHub PR: err=%s\n%v\n", err, resp)
+					} else {
+						ghPrClientDetails.PrLogger.Debugf("PR %v labeled\n%+v", *eventPayload.PullRequest.Number, prLables)
+					}
+					// TODO Auto-merge PRs with no changes(optional)
```

Comment:
No, because I only want to auto-merge PRs that causes changes to a cluster.
GH auto-merge can only rely on success/failure of "checks"(AFAIK)  but I wouldn't want to fail a check because it has diff - I just want a human to approve it :)